smash-pumpkins.lic

```diff
@@ -1,48 +1,67 @@
-custom_require.call(%w[events common-travel common])
+=begin
+  Documentation: https://elanthipedia.play.net/Lich_script_repository#smash-pumpkins
+=end
+
+custom_require.call(%w[events common-items common-travel common])
 
-UserVars.last_darkbox = Room.current.id
 Flags.add('pumpkin-leeches', 'only to rain down around you afterwards')
 
 junk_list = get_settings.pumpkin_junk.map { |x| /\b#{x}/i }
 
 DRCT.walk_to(10_570)
 loop do
   break if checkbleeding
-  break if checkhealth < 75
+  break if checkhealth < 50
 
   until checkright || checkleft
-    fput('stand') unless checkstanding
-    if /injured/ =~ DRC.bput('grab vat', 'roundtime', 'You are too injured', 'Perhaps that would work better')
-      DRCT.walk_to(19_104)
+    DRC.fix_standing
+    case DRC.bput('grab vat', 'roundtime', 'You are too injured', 'Perhaps that would work better', 'when you have enough coins')
+    when 'You are too injured'
+      DRCT.walk_to(10_341)
       exit
+    when 'when you have enough coins'
+      DRCT.walk_to(10_358)
+      case DRC.bput("withdraw 3 platinum", 'we are not lending money at this time', 'The clerk counts out')
+      when 'we are not lending money at this time'
+        exit
+      end
+      DRCT.walk_to(10_570)
     end
     pause 1
     waitrt?
   end
 
   DRCT.walk_to(10_569)
 
-  DRC.bput('break my pumpkin', 'wipe off pumpkin residue')
+  case DRC.bput('break my pumpkin', 'wipe off pumpkin residue', 'Break what', 'need a free hand', 'You should hold')
+  when 'need a free hand'|| 'Break what'
+    DRCI.stow_hands
+  end
   pause
 
   DRCT.walk_to(10_570)
   while thing = DRC.left_hand || DRC.right_hand
     case thing
     when *junk_list
-      fput("put #{thing} in bucket")
+      DRC.bput("put #{thing} in bucket", 'You drop')
```

Comment:
Does `dispose_trash` do what you need?

Reply:
Tried changing it to DRCI.dispose_trash(thing) but it just does this
[smash-pumpkins]>put my tattered scarf in bucket
You drop a tattered scarf in a dented metal bucket.
>
[smash-pumpkins]>stow tattered scarf
Stow what?  Type 'STOW HELP' for details.
...
[smash-pumpkins: checked against [/You put/i, /to fit in the/i]]
[smash-pumpkins: for command stow skeleton earrings]

Reply:
That looks like what I'd expect to happen. Line 46 does a 'put' and line 47 does a 'stow'.

Reply:
I see the problem, I removed the else by mistake, they weren't both meant to be run. Fixed up now.

Reply:
Cool. I'm still in favor of `dispose_trash`, but it isn't a big deal.